From ff86c42c8614828be77810a50202e1f771b35ff3 Mon Sep 17 00:00:00 2001
From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Mon, 21 Oct 2013 21:29:23 +0200
Subject: [PATCH] analyze: set white backgound

In programs like eog and gimp the transparant background did not
look very good.

https://bugs.freedesktop.org/show_bug.cgi?id=70720
---
 src/analyze/systemd-analyze.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/analyze/systemd-analyze.c b/src/analyze/systemd-analyze.c
index a4f15eb..9a69795 100644
--- a/src/analyze/systemd-analyze.c
+++ b/src/analyze/systemd-analyze.c
@@ -507,6 +507,7 @@ static int analyze_plot(DBusConnection *bus) {
         /* style sheet */
         svg("<defs>\n  <style type=\"text/css\">\n    <![CDATA[\n"
             "      rect       { stroke-width: 1; stroke-opacity: 0; }\n"
+            "      rect.background   { fill: rgb(255,255,255); }\n"
             "      rect.activating   { fill: rgb(255,0,0); fill-opacity: 0.7; }\n"
             "      rect.active       { fill: rgb(200,150,150); fill-opacity: 0.7; }\n"
             "      rect.deactivating { fill: rgb(150,100,100); fill-opacity: 0.7; }\n"
@@ -528,6 +529,7 @@ static int analyze_plot(DBusConnection *bus) {
             "      text.sec   { font-size: 10px; }\n"
             "    ]]>\n   </style>\n</defs>\n\n");
 
+        svg("<rect class=\"background\" width=\"100%%\" height=\"100%%\" />\n");
         svg("<text x=\"20\" y=\"50\">%s</text>", pretty_times);
         svg("<text x=\"20\" y=\"30\">%s %s (%s %s) %s</text>",
             isempty(osname) ? "Linux" : osname,
